rocAL/source/decoders/image/rocjpeg_decoder.cpp

```diff
@@ -0,0 +1,386 @@
+    RocJpegChromaSubsampling subsampling;
+    uint8_t num_components;
+    uint32_t widths[4] = {};
+    uint32_t heights[4] = {};
+    uint32_t channels_size = 0;
+    uint32_t channel_sizes[4] = {};
+
+    uint32_t max_widths[4] = {static_cast<uint32_t>(max_decoded_width), 0, 0, 0};
+    uint32_t max_heights[4] = {static_cast<uint32_t>(max_decoded_height), 0, 0, 0};
+
+    switch(desired_decoded_color_format) {
+        case Decoder::ColorFormat::GRAY:
+            _decode_params.output_format = ROCJPEG_OUTPUT_Y;
+            _num_channels = 1;
+            break;
+        case Decoder::ColorFormat::RGB:
+        case Decoder::ColorFormat::BGR:
+            _decode_params.output_format = ROCJPEG_OUTPUT_RGB;
+            _num_channels = 3;
+            break;
+    };
+
+    if (rocJpegStreamParse(reinterpret_cast<uint8_t*>(input_buffer), input_size, _rocjpeg_streams[index]) != ROCJPEG_STATUS_SUCCESS) {
+        return Status::HEADER_DECODE_FAILED;
+    }
+    if (rocJpegGetImageInfo(_rocjpeg_handle, _rocjpeg_streams[index], &num_components, &subsampling, widths, heights) != ROCJPEG_STATUS_SUCCESS) {
+        return Status::HEADER_DECODE_FAILED;
+    }
+
+    if (widths[0] < 64 || heights[0] < 64) {
+        return Status::CONTENT_DECODE_FAILED;
+    }
+
+    std::string chroma_sub_sampling = "";
+    GetChromaSubsamplingStr(subsampling, chroma_sub_sampling);
+    if (subsampling == ROCJPEG_CSS_411 || subsampling == ROCJPEG_CSS_UNKNOWN) {
+        return Status::UNSUPPORTED;
+    }
+
+    if (width) *width = widths[0];
+    if (height) *height = heights[0];
+    uint scaledw = widths[0], scaledh = heights[0];
+    // Scaling to be performed if width/height is greater than max decode width/height
+    if (widths[0] > max_decoded_width || heights[0] > max_decoded_height) {
+        for (unsigned j = 0; j < _num_scaling_factors; j++) {
+            scaledw = (((widths[0]) * _scaling_factors[j].num + _scaling_factors[j].denom - 1) / _scaling_factors[j].denom);
+            scaledh = (((heights[0]) * _scaling_factors[j].num + _scaling_factors[j].denom - 1) / _scaling_factors[j].denom);
+            if (scaledw <= max_decoded_width && scaledh <= max_decoded_height)
+                break;
+        }
+    }
+    // If scaled width is different than original width and height, update max dims with the original width and height, to be used for decoding
+    if (scaledw != widths[0] || scaledh != heights[0]) {
+        _resize_batch = true;   // If the size of any image in the batch is greater than max size, resize the complete batch
+        max_widths[0] = (widths[0] + 8) &~ 7;
+        max_heights[0] = (heights[0] + 8) &~ 7;
+    }
+
+    if (GetChannelPitchAndSizes(_decode_params.output_format, subsampling, max_widths, max_heights, channels_size, _output_images[index], channel_sizes)) {
```

Comment:
Please change `_decode_params.output_format` to `_decode_params[index]`

Reply:
Done